lang/java/avro/src/test/java/org/apache/avro/io/TestBinaryDecoder.java

```diff
@@ -317,7 +320,7 @@ private void validateInputStreamSkips(InputStream test, InputStream check) throw
   public void testBadIntEncoding() throws IOException {
     byte[] badint = new byte[5];
     Arrays.fill(badint, (byte) 0xff);
-    Decoder bd = factory.binaryDecoder(badint, null);
+    Decoder bd = this.newDecoder(badint);
```

Comment:
To clarify the purpose of the JIRA: this test only checks that `binaryEncoder` correctly deals with badly encoded INTs, and it does it twice.  One of those times, `directBinaryEncoder` should be called!

Reply:
Change, all methods now test DirectBinaryDecoder, then BinaryDecoder. We can see that the behavior sometimes differs  with the type of exceptions that are thrown.